CCSocialShare/android/java/src/org/kawaz/socialshare/SocialShare.java

```diff
@@ -26,25 +42,40 @@
     final static String TWITTER_PACKAGE_NAME = "com.twitter.android";
     final static String FACEBOOK_PACKAGE_NAME = "com.facebook.katana";
     final static int REQUEST_CODE = 100;
-    static public void postToTwitter(String message, String imagePath) {
-        String url = "http://twitter.com/share?text=" + message;
-        Intent intent = new Intent(Intent.ACTION_SEND, Uri.parse(url));
-        intent.putExtra(Intent.EXTRA_TEXT, message);
-        if (!imagePath.equals("")) {
-            try {
-                Uri originalImageUri = Uri.parse(imagePath);
-                String ext = getExtension(originalImageUri);
-                Uri readableFileUri = saveImageToExternalDirectory(originalImageUri);
-                if (readableFileUri != null) {
-                    intent.setType("image/" + ext);
-                    intent.putExtra(Intent.EXTRA_STREAM, readableFileUri);
+    private static TwitterAuthClient authClient = null;
+    private static File imageFile = null;
+
+    static private void postToTwitter(final String message, final String imagePath) {
+        TwitterConfig config = new TwitterConfig.Builder(Cocos2dxActivity.getContext()).debug(true).build();
+        Twitter.initialize(config);
+        Context context = Cocos2dxActivity.getContext();
+        authClient = new TwitterAuthClient();
+        authClient.authorize((Activity) context, new Callback<TwitterSession>() {
+            @Override
+            public void success(Result<TwitterSession> result) {
+                Context context = Cocos2dxActivity.getContext();
+                TwitterSession session = result.data;
+                ComposerActivity.Builder builder = new ComposerActivity.Builder(context).session(session).text(message);
+                if (!imagePath.equals("")) {
+                    try {
+                        Uri originalImageUri = Uri.parse(imagePath);
+                        Uri readableFileUri = saveImageToExternalDirectory(originalImageUri);
+                        if (readableFileUri != null) {
+                            builder.image(readableFileUri);
+                        }
+                    } catch (Exception e) {
+                        Log.d(TAG, e.getMessage());
```

Comment:
It is not suitable to use `Log.d`
I think you should use `Log.e` to display errors